lib/deep_slice.rb

```diff
@@ -13,9 +13,19 @@ def deep_slice(*keys)
             recursive_keys = key[sub_key]
             result[sub_key] = \
               if self[sub_key].is_a?(Array)
-                self[sub_key].map { |val| val.deep_slice(*[recursive_keys].flatten) }
+                self[sub_key].map { |val|
```

Comment:
Might be a bit tidier if written like `val.deep_slice(*[recursive_keys].flatten) unless val.nil?`